Randomize sal

saljaffal opened the pull request:
Created a randomize function in order to return  10 random tv shows. 

1.  The function will call setRandomNumber() which will set a random number between 0 and 216 (max number of pages of the API URL.
2. Then that number will be passed onto the URL and it would return an array of max 250 objects.
3. Not all 250 object will be added to our allRandomShows array as it will return undefined for any TV show that is not in English.
4. This resulting array will be passed onto a .filter method to exclude all the undefined TV Shows which would finally return an array of all shows in English.
5. This array would then be sliced at a random point (firstRandomShow) which would then extend until the next 10 tv shows (lastRandomShow)

Commit:
Added a randomize function in the UserSelectTv component and added a language deconstructor to the tvShow constructor in order to filter out TV Shows that aren't in English

Commit:
Merge branch 'main' of https://github.com/nextBinge-squad/nextBinge into randomize-sal

Commit:
adding required to text input field

Commit:
Merge branch 'main' of https://github.com/nextBinge-squad/nextBinge into randomize-sal

Commit:
changed a comment on one of the functions

Commit:
merged the change of the button to the UserSelectTv

LeonBaram commented:
Hey, this looks pretty good for the most part! I have a couple suggestions about the filtering part though.

Currently it looks like this:

```
const allRandomShows = data.map((show) => {

  if (show.language === "English") {  
    return new tvShow(show)
  }
  else { }
})

const filterAllRandomShows = allRandomShows.filter((show) => {
  return show !== undefined;
});
const firstRandomShow = setRandomNumber(0, (filterAllRandomShows.length - 10));
const lastRandomShow = firstRandomShow + 10;

setRandomShows(filterAllRandomShows.slice(firstRandomShow, lastRandomShow));
```
My first suggestion is to put the `show.language` test inside of the `filter()` operation itself. For example:
```
const englishShows = data.filter((show) => show.language === "English");
```
My second suggestion is, it seems like `setRandomNumber()` has a lot of math going on, versus the "classic" floor+random approach. I'm picturing something like this:
```
const range = 10;

const englishShows = data.filter((show) => show.language === "English");

const randIndex = Math.floor(Math.random() * (englishShows.length - range));

setRandomShows(englishShows.splice(randIndex, randIndex+range);
```
Obviously the variable names don't matter, but let me know what you think about the overall structure change!

Commit:
changed the random number function to a simpler equation and removed the if statement for allRandomShows and just applied the filter direclty to compare for english language

saljaffal commented:
Hey Leon, so I changed the random number function to a simpler method and since it is called twice I thought it's better to have a separate function for it. and I removed the if statements and just applied the filter directly. Lemme know if you have any more suggestions
